tests/D2L.CodeStyle.Analyzers.Test/Specs/OnlyVisibleToAnalyzer.cs

```diff
@@ -261,3 +261,65 @@ public static class DisallowedTypeOfCaller {
 		}
 	}
 }
+
+// ===============================================================================
+
+namespace Targets {
+
+	public static class GenericCallerTypes {
+
+		[OnlyVisibleToType( typeof( TestCases.AllowedGenericCaller<> ) )]
+		public static void VisibleByTypeOf() { }

```

Comment:
You'd want an `[OnlyVisibleToType( typeof( TestCases.AllowedGenericCaller<int> ) )]` example to demonstrate the 			`GetTypeOrUnboundGenericType` is functioning, no?

Reply:
It doesn't actually make sense to define that.   At compilation time it's always T calling you.

Reply:
Like `TestCases.AllowedGenericCaller<int>` and `TestCases.AllowedGenericCaller<string>` are both defined by the same generic type `TestCases.AllowedGenericCaller<>`.

Reply:
If anything, I can try and block that.

Reply:
Right sorry, that's what I meant by IsConsructedGenericType. I agree it's improper, but figured should either make it work or raise a diagnostic for using the API wrong.

For a diagnostic I was expecting `if type != type.OriginalDefinition` or something.

Reply:
Added a test case to show it doesn't work:

https://github.com/Brightspace/D2L.CodeStyle/pull/843/commits/74032264ec4808c6ba72b75e738d447a4351e728

Reply:
I'm not really sure if it's worth scanning attributes to solve this.  It won't compile, which is the main point.  I would imagine most people using these attributes know how to specify the generic type definition.

Reply:
Going to start with just documentation.